src/utils/test/qa_c_wrapper.cpp

```diff
@@ -0,0 +1,206 @@
+#include <boost/ut.hpp>
+
+#include "../include/c_resource.hpp"
+
+#include <atomic>
+#include <fmt/format.h>
+
+struct S {
+    int  counter = 0; // instance counter
+
+    void operator++(int) noexcept { ++counter; }
+    void operator--(int) noexcept { --counter; }
+
+    auto operator<=>(const S &) const = default;
+};
+extern "C" {
+inline static S cApiRessource; // NOSONAR acts as a global resource that needs to be managed
+// test functions
+S *conS1() {
+    cApiRessource++;
+    return &cApiRessource;
+}; // API schema 1: pointer-type return value
+void desS1(S *ressource) {
+    ressource->counter--;
+    ressource = nullptr;
+} // API schema 1: value as input parameter
+void conS2(S **ressource) {
+    if (ressource) {
+        *ressource = &cApiRessource; // Point to the global instance
+        (*ressource)->counter++;
+    }
+} // API schema 2: reference as input-output parameter
+void desS2(S **ressource) {
+    (*ressource)->counter--;
+    *ressource = nullptr;
+} // API schema 2: reference as input-output parameter
+}
+
+const static boost::ut::suite c_wrapper_API = [] {
+    using w1 = stdex::c_resource<S *, conS1, desS1>;
+    using w2 = stdex::c_resource<S *, conS2, desS2>;
+
+    static_assert(std::is_nothrow_default_constructible_v<w1>);
+    static_assert(!std::is_copy_constructible_v<w1>);
+    static_assert(std::is_nothrow_move_constructible_v<w1>);
+    static_assert(!std::is_copy_assignable_v<w1>);
+    static_assert(std::is_nothrow_move_assignable_v<w1>);
+    static_assert(std::is_nothrow_destructible_v<w1>);
+    static_assert(std::is_nothrow_swappable_v<w1>);
+
+    static_assert(std::is_nothrow_default_constructible_v<w2>);
+    static_assert(!std::is_copy_constructible_v<w2>);
+    static_assert(std::is_nothrow_move_constructible_v<w2>);
+    static_assert(!std::is_copy_assignable_v<w2>);
+    static_assert(std::is_nothrow_move_assignable_v<w2>);
+    static_assert(std::is_nothrow_destructible_v<w2>);
+    static_assert(std::is_nothrow_swappable_v<w2>);
+};
+
+const static boost::ut::suite c_wrapper_functional_c_api_schema12 = [] {
```

Comment:
Is there a test for non-destructible c_resource?

Reply:
no, but that would/could be handled via an empty destructor lambda function.